### PR TITLE
Add a URL checker for docs in Travis. [needs revision]

### PR DESCRIPTION
**Pull request by jsha**

### CODE REVIEW

File: tests/url-checker.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+errors = []
+threads = []
+urls = {}
+for filename in sys.argv[1:]:
+    contents = open(filename).read()
+    for url in re.findall("https?://[^\s]+", contents):
+        url = url.rstrip(">.,)`_\"")
+        url = re.sub("#.*", "", url)
+        if url not in urls:
+            print url
```
Comment (SwartzCr) [inline tests/url-checker.py]: this might be causing py3+ to break

File: tests/url-checker.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    try:
+        r = requests.get(url)
+        if r.status_code > 500: # Do one retry.
+            time.sleep(1)
+            r = requests.get(url)
+        if r.status_code != 200:
+            return ("Status code %d fetching %s (mentioned in %s)" %
+                (r.status_code, url, filename))
+    except Exception, e:
```
Comment (SwartzCr) [inline tests/url-checker.py]: oh, this is causing py 3+ to break

**Comment by jsha** [comment]
Fixed the test issues, now passes.

**Comment by jsha** [comment]
@pde @SwartzCr: We now have some out-of-date links in the docs:

```
Status code 404 fetching https://travis-ci.org/letsencrypt/letsencrypt.svg?branch=master (mentioned in README.rst)
Status code 404 fetching https://google-styleguide.googlecode.com/svn/trunk/pyguide.html (mentioned in docs/contributing.rst)
```

Please fix up so we can merge this PR.

**Comment by SwartzCr** [comment]
lint is now chocking on 
```
urlcheck runtests: commands[0] | bash -c git ls-files '*.md' '*.rst' '*.txt' | xargs python tests/url-checker.py
Problem fetching https://acme- (mentioned in docs/cli-help.txt): <urlopen error [Errno -2] Name or service not known>
ERROR: InvocationError: "/bin/bash -c git ls-files '*.md' '*.rst' '*.txt' | xargs python tests/url-checker.py"
```
which comes from line 127 of cli-help.txt
somehow URL checker should be ignoring that line

**Comment by jsha** [comment]
Going to close this out for now. Since we've completed the rename transition I think the risk of broken links is lower.

**Pull request closed without merging**